src/node/server/test-server.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+'use strict';
+
+/* eslint-env node */
+
+const express = require('express');
+
+class TestServer {
+  constructor(addDefaultRoutes) {
+    if (
+      typeof addDefaultRoutes !== 'undefined' &&
+      typeof addDefaultRoutes !== 'boolean'
+    ) {
+      throw new Error('addDefaultRoutes must be a boolean value');
+    }
+
+    this._server = null;
+    this._app = express();
+    this._useDefaults = false;
+
+    if (typeof addDefaultRoutes === 'undefined' || addDefaultRoutes) {
+      this._useDefaults = true;
+      this.addDefaultRoutes();
+    }
+  }
+
+  addDefaultRoutes() {
+    // If the user tries to go to the root of the server, redirect them
+    // to the browser test path
+    this._app.get('/', function(req, res) {
+      res.redirect('/test/browser-tests/');
+    });
+  }
+
+  getExpressApp() {
+    return this._app;
+  }
+
+  startServer(path, portNumber, host) {
+    if (this._server) {
+      this._server.close();
+    }
+
+    // 0 will pick a random port number
+    if (typeof portNumber === 'undefined') {
```

Comment:
This and a number of other places in the code would be good candidates for [default parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Default_parameters), unless you're not transpiling and are concerned about older versions of `node`.